File: ide/xcode/L0104HelloCPP/L0104HelloCPP/Man.cpp
//
//  Man.cpp
//  L0104HelloCPP
//
//  Created by plter on 13-10-26.
//  Copyright (c) 2013年 plter. All rights reserved.
//

#include "Man.h"

namespace eoe {
    
    Man::Man(int _age,const char *_name):People(_age,_name){
        
    }
    
    void Man::love(eoe::Woman &woman){
        std::cout<<getName()<<" loves a woman called "<<woman.getName()<<"\n";
    }
    
    void Man::sayHi(){
        std::cout<<"Man class\n"<<"Hello "<<getName()<<"\n";
    }
}

File: ide/ldt/L0606OOLua1/src/Man.lua
require("People")

function Man_new(name)
	
	local self = People_new(name)
	
	self.sayHi = function (this)
		print("Man say : Hi "..this.name)
	end
	
	self.sayHello = function (this)
		print("Man say : Hello "..this.name)
	end
	
	return self
end